### PR TITLE
Use named placeholders in translated strings

### PR DESCRIPTION
**Pull request by phw**
# Summary

* This is a…
  * [ ] Bug fix
  * [ ] Feature addition
  * [x] Refactoring
  * [ ] Minor / simple change (like a typo)
  * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

Some translation strings contain multiple placeholders like 

```python
"%s (%i release)" % (name, size)
```

This requires translations to maintain the order of the placeholders. Some strings might be better or only properly translated in a different order.

# Solution

Use named placeholders.

For multiple placeholders this allows changing the order in translation. The name also generally makes it easier for the translator to understand the string.

# Action

After this has been merged I'll update the translation template and translations themselves. Actually we should be able to fully keep existing translations by changing the placeholders only.

Weblate has a check "[Multiple unnamed variables](https://translations.metabrainz.org/checks/unnamed_format/)" to check this condition.

### CODE REVIEW

File: picard/ui/options/maintenance.py [hunk 1]
```diff
@@ -124,11 +124,10 @@ def load(self):
         orphan_options = file_options.difference(current_options)
 
         self.ui.option_counts.setText(
-            _("The configuration file currently contains %d option settings, %d which are unused.") % (
-                len(file_options),
-                len(orphan_options)
-            )
-        )
+            _("The configuration file currently contains %(totalcount)d option settings, %(unusedcount)d which are unused.") % {
```
Comment (zas) [inline picard/ui/options/maintenance.py:127]: The string is pretty long, may be wrap:

```suggestion
            _("The configuration file currently contains %(totalcount)d option settings, "
                "%(unusedcount)d which are unused.") % {
```

**Review by zas — CHANGES_REQUESTED** [review]
Very good move, it will ease translator's job.

File: picard/ui/infodialog.py [hunk 2]
```diff
@@ -207,14 +207,20 @@ def _display_artwork(self, images, col):
                     pixmap.loadFromData(data)
                     item.setToolTip(
                         _("Double-click to open in external viewer\n"
-                        "Temporary file: %s\n"
-                        "Source: %s") % (image.tempfile_filename, image.source))
+                        "Temporary file: %(tempfile)s\n"
+                        "Source: %(sourcefile)s") % {
+                            'tempfile': image.tempfile_filename,
+                            'sourcefile': image.source,
+                        })
```
[the remark this answers is not in the document]
Comment (phw) [reply picard/ui/infodialog.py:214]: That makes a bit of a mess for translation, as in your example above the 


```
"Temporary file: %(tempfile)s\n"
"Source: %(sourcefile)s"
```

would also have to be translated. Then the full string gets build together, which is often intransparent for translators and also does not always give the best results. Also I find the resulting message:


> Missing temporary file
> Temporary file: foo
> Source: bar

not so elegant. Existing is better:

> Missing temporary file: foo
> Source bar:

We could move the dict

```
{
    'tempfile': image.tempfile_filename,
    'sourcefile': image.source,
}
```

outside. But then  the subsitutions are defined quite a bit away from the usage.

Really this function could need some cleanup, but not here when updating the strings.

File: picard/pluginmanager.py [hunk 3]
```diff
@@ -214,7 +214,9 @@ def plugin_error(self, name, error, *args, **kwargs):
         passed to the function given via the `log_func` keyword argument
         (default: log.error) and as the error message to the `plugin_errored`
         signal."""
-        error = error % args
+        params = kwargs.get('params', args)
```
[the remark this answers is not in the document]
Comment (phw) [reply picard/pluginmanager.py:221]: Done

File: picard/ui/options/genres.py [hunk 4]
```diff
@@ -153,7 +153,8 @@ def update_test_genres_filter(self):
         # FIXME: very simple error reporting, improve
         self.ui.label_test_genres_filter_error.setText(
             "\n".join(
-                [_("Error line %d: %s") % (lineno + 1, error) for lineno, error in tagfilter.errors.items()]
+                _("Error line %(lineno)d: %(error)s") % {'lineno': lineno + 1, 'error': error}
+                for lineno, error in tagfilter.errors.items()
```
[the remark this answers is not in the document]
Comment (phw) [reply picard/ui/options/genres.py]: Good idea. I called the function `format_errors`

File: picard/ui/options/maintenance.py (hunk 1 above)
Comment (phw) [reply picard/ui/options/maintenance.py]: The entire block of code looks better when splitted

**Comment by phw** [comment]
@zas I'll merge it once I'm done with populating the constants translations again.

**Pull request merged**